extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/K8sTaskAdapter.java

```diff
@@ -88,13 +88,17 @@ public Job fromTask(Task task, PeonCommandContext context) throws IOException
   {
     String myPodName = System.getenv("HOSTNAME");
     Pod pod = client.executeRequest(client -> client.pods().inNamespace(config.namespace).withName(myPodName).get());
-    return createJobFromPodSpec(pod.getSpec(), task, context);
+    PodSpec podSpec = pod.getSpec();
+    massageSpec(config, podSpec);
+    return createJobFromPodSpec(podSpec, task, context);
   }
 
   @Override
   public Task toTask(Pod from) throws IOException
   {
-    List<EnvVar> envVars = from.getSpec().getContainers().get(0).getEnv();
+    PodSpec podSpec = from.getSpec();
+    massageSpec(config, podSpec);
+    List<EnvVar> envVars = podSpec.getContainers().get(0).getEnv();
```

Comment:
Can the podspec have zero containers? Do we need an empty check here in that case?

Reply:
a peon podSpec can't have 0 containers.  To create the peon PodSpec we get the druid container from the overlord podSpec and work off that.